lib/components/map/gtfs-rt-vehicle-overlay.js

```diff
@@ -0,0 +1,218 @@
+import { transit_realtime as transitRealtime } from 'gtfs-realtime-bindings'
+import L from 'leaflet'
+import TransitVehicleOverlay from '@opentripplanner/transit-vehicle-overlay'
+import PropTypes from 'prop-types'
+import React from 'react'
+import ReactDOMServer from 'react-dom/server'
+import { FeatureGroup, MapLayer, Marker, Polyline } from 'react-leaflet'
+import styled from 'styled-components'
+
+export const VehicleShape = styled.span`
+  background-color: #${props => props.route?.route_color || '000000'};
+  border: 1px solid #000000;
+  border-radius: 50%;
+  color: #${props => props.route?.route_text_color || 'ffffff'};
+  display: block;
+  height: 24px;
+  left: -6px;
+  line-height: 24px;
+  margin-left: auto;
+  margin-right: auto;
+  position: relative;
+  text-align: center;
+  top: -6px;
+  width: 24px;
+`
+
+/**
+ * Convert GTFS-rt entities to vehicle location from OTP-UI.
+ * @param vehicle The GTFS-rt vehicle position entity to convert.
+ * @param routes Optional GTFS-like routes list.
+ */
+function gtfsRtToTransitVehicle (gtfsRtVehicle, routes) {
+  const { position, stopId, timestamp, trip, vehicle } = gtfsRtVehicle.vehicle
+  let routeShortName
+  if (routes.length !== 0) {
+    routeShortName = routes.find(r => r.route_id === trip.routeId)?.route_short_name
+  }
+  return {
+    heading: position.bearing,
+    id: vehicle.id,
+    lat: position.latitude,
+    lon: position.longitude,
+    reportDate: new Date(parseInt(timestamp, 10)).toLocaleString(),
+    routeShortName,
+    routeType: 'BUS',
```

Comment:
Is there a way to pass in a function that can calculate the routeType here instead of hard-coding it like this?

Reply:
Yes, reusing one that already does in https://github.com/opentripplanner/otp-react-redux/pull/401/commits/4348cdb2c20d2658d9b4f113dd15e18d3b110939.